[CI] CrossClusterCancellationIT testCancelSkipUnavailable failing

**Build Scans:**
- [elasticsearch-periodic #10671 / release-tests](https://gradle-enterprise.elastic.co/s/6l3q6xpt6ga5i)
- [elasticsearch-pull-request #92657 / part-3](https://gradle-enterprise.elastic.co/s/lwyrd7yswojuw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.CrossClusterCancellationIT.testCancelSkipUnavailable" -Dtests.seed=3E3AB38861B520D9 -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=sg-CF -Dtests.timezone=Cuba -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.CrossClusterCancellationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testCancelSkipUnavailable'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 2 failures in test testCancelSkipUnavailable (0.8% fail rate in 258 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -183,9 +183,6 @@ tests:
 - class: org.elasticsearch.aggregations.bucket.AggregationReductionCircuitBreakingIT
   method: testCBTrippingOnReduction
   issue: https://github.com/elastic/elasticsearch/issues/134667
-- class: org.elasticsearch.xpack.esql.action.CrossClusterCancellationIT
-  method: testCancelSkipUnavailable
-  issue: https://github.com/elastic/elasticsearch/issues/134865
 - class: org.elasticsearch.reservedstate.service.RepositoriesFileSettingsIT
   method: testSettingsApplied
   issue: https://github.com/elastic/elasticsearch/issues/126748

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterCancellationIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterCancellationIT.java
@@ -19,6 +19,8 @@ import org.elasticsearch.common.transport.TransportAddress;
 import org.elasticsearch.compute.operator.DriverTaskRunner;
 import org.elasticsearch.compute.operator.exchange.ExchangeService;
 import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
 import org.elasticsearch.tasks.TaskCancelledException;
 import org.elasticsearch.tasks.TaskInfo;
 import org.elasticsearch.transport.TransportService;
@@ -42,6 +44,8 @@ import static org.hamcrest.Matchers.instanceOf;
 
 public class CrossClusterCancellationIT extends AbstractCrossClusterTestCase {
     private static final String REMOTE_CLUSTER = "cluster-a";
+
+    private static final Logger LOGGER = LogManager.getLogger(CrossClusterCancellationIT.class);
 
     @Override
     protected List<String> remoteClusterAlias() {
@@ -87,17 +91,22 @@ public class CrossClusterCancellationIT extends AbstractCrossClusterTestCase {
         String stats = randomStats();
         EsqlQueryRequest request = syncEsqlQueryRequest("FROM *:test | " + stats + " total=sum(const) | LIMIT 1").pragmas(randomPragmas());
         PlainActionFuture<EsqlQueryResponse> requestFuture = new PlainActionFuture<>();
+        LOGGER.info("Executing query {}", request);
         client().execute(EsqlQueryAction.INSTANCE, request, requestFuture);
+        LOGGER.info("Waiting for query to start");
         assertTrue(SimplePauseFieldPlugin.startEmitting.await(30, TimeUnit.SECONDS));
+        LOGGER.info("Query started, checking tasks");
         List<TaskInfo> rootTasks = new ArrayList<>();
         assertBusy(() -> {
             List<TaskInfo> tasks = client().admin().cluster().prepareListTasks().setActions(EsqlQueryAction.NAME).get().getTasks();
             assertThat(tasks, hasSize(1));
             rootTasks.addAll(tasks);
         });
+        LOGGER.info("Query started, now cancelling root task");
         var cancelRequest = new CancelTasksRequest().setTargetTaskId(rootTasks.get(0).taskId()).setReason("proxy timeout");
         client().execute(TransportCancelTasksAction.TYPE, cancelRequest);
         try {
+            LOGGER.info("Waiting for drivers to be cancelled");
             assertBusy(() -> {
                 List<TaskInfo> drivers = client(REMOTE_CLUSTER).admin()
                     .cluster()

